### PR TITLE
Use reusable workflows to split linux and windows builds

### PR DESCRIPTION
**Pull request by jprendes**
This is a continuation of #267 and further reduces the PR runtime to \~8min

### CODE REVIEW

File: .github/workflows/action-fmt.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    inputs:
+      os:
+        required: true
+        type: string
+
+jobs:
+  fmt:
+    runs-on: ${{ inputs.os }}
+    steps:
+      - name: Free Disk Space (Ubuntu)
```
Comment (jsturtevant) [inline .github/workflows/action-fmt.yml:14]: do the linting and building jobs need the clean up?  might make them faster without?

Comment (jprendes) [reply .github/workflows/action-fmt.yml:14]: I believe that by sharing artifacts we might not need the cleanup anywhere

Comment (jprendes) [reply .github/workflows/action-fmt.yml]: By the end, the job with the highest disk usage still had 13Gb free (started with 19 Gb free), so I think it's ok to skip the space reclaiming step.

**Review by Mossaka — APPROVED** [review]
LGTM! Thanks for reducing the CI time significantly!

**Pull request merged**